File: js/leetcode/_7.js
// 1 <= word1.length, word2.length <= 103
// 1 <= word1[i].length, word2[i].length <= 103
// 1 <= sum(word1[i].length), sum(word2[i].length) <= 103
// word1[i] and word2[i] consist of lowercase letters.
/**
 * @param {string[]} word1
 * @param {string[]} word2
 * @return {boolean}
 */
export default function (word1, word2) {
  return word1.join("") === word2.join("");
}

File: src/leetcode/_7.rs
#![allow(dead_code)]
struct Solution;
impl Solution {
    pub fn array_strings_are_equal(word1: Vec<String>, word2: Vec<String>) -> bool {
        return word1.join("") == word2.join("");
    }
}
#[cfg(test)]
mod test {
    use super::*;

    #[test]
    fn basic_test() {
        let word1 = vec!["ab".to_string(), "c".to_string()];
        let word2 = vec!["a".to_string(), "bc".to_string()];
        let result = Solution::array_strings_are_equal(word1, word2);
        let expected = true;

        assert_eq!(result, expected);
    }
}
